invenio_rdm_records/services/communities/service.py

```diff
@@ -6,24 +6,100 @@
     """
 
+    @property
+    def indexer(self):
+        """Factory for creating an indexer instance."""
+        return self.config.indexer_cls(
+            # the routing key is mandatory in the indexer constructor since
+            # it is afterwards passed explicitly to the created consumers
+            # and producers. this means that it is not strictly necessary to
+            # pass it to the queue constructor. however, it is passed for
+            # consistency (in case the queue is used by itself) and to help
+            # entity declaration on publish.
+            queue=LocalProxy(
+                lambda: Queue(
+                    self.config.indexer_queue_name,
+                    exchange=current_app.config["INDEXER_MQ_EXCHANGE"],
+                    routing_key=self.config.indexer_queue_name,
+                )
+            ),
+            routing_key=self.config.indexer_queue_name,
+            record_cls=self.config.record_cls,
+            record_to_index=self.record_to_index,
+            record_dumper=self.config.index_dumper,
+        )
+
+    def record_to_index(self, record):
+        """Function used to map a record to an index."""
+        return record.index._name
+
+    @property
+    def schema(self):
+        """Returns the data schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.schema)
+
+    @property
+    def record_cls(self):
+        """Factory for creating a record class."""
+        return self.config.record_cls
+
     @unit_of_work()
     def create(self, identity, data, record, uow=None):
         """Include the record in the given communities."""
         pass
 
+    def _remove(self, community_id, record):
+        """Remove a community from the record."""
+        if community_id not in record.parent.communities.ids:
+            raise RecordCommunityMissing(record.id, community_id)
+
+        record.parent.communities.remove(community_id)
+
     @unit_of_work()
-    def delete(self, identity, id_, revision_id=None, uow=None):
+    def delete(self, identity, record_id, data, revision_id=None, uow=None):
         """Remove communities from the record."""
-        pass
+        record = self.record_cls.pid.resolve(record_id)
+        self.require_permission(identity, "remove_community", record=record)
+
+        valid_data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=True,
+        )
+        communities = valid_data["communities"]
+        if self.config.max_number_of_removals < len(communities):
+            raise ExceedMaxNumberOfCommunities(self.config.max_number_of_removals)
+
+        for community in communities:
+            community_id = community["id"]
+            try:
+                self._remove(community_id, record)
+                uow.register(RecordCommitOp(record.parent))
+                self.indexer.index(record)
```

Comment:
RIght! Thanks.
Note that we are committing the record.parent but indexing the record so I guess we would need to:
```
uow.register(RecordCommitOp(record.parent))
uow.register(RecordCommitOp(record, indexer=self.indexer))
```
because the record indexer would not work with the parent record.